modules/library/render/src/test/java/org/geotools/renderer/lite/RepeatedLabelTest.java

```diff
@@ -77,7 +90,36 @@ public void testRepeatedLabels(String styleName) throws Exception {
 
         BufferedImage image = RendererBaseTest.renderImage(renderer, bounds, null, 500, 500);
         File expected = new File("src/test/resources/org/geotools/renderer/lite/test-data/" + styleName + ".png");
-        int tolerance = 300;
+        int tolerance = 1000;
+        ImageAssert.assertEquals(expected, image, tolerance);
+    }
+    
+    @Test
+    public void testLabelSquareBorders() throws Exception {
+        testRepeatedLabelsPolygonBorder(square, "repeatedLabelsAlongLine", "poly", 1000);
+    }
+    
+    @Test
+    public void testLabelSquareBordersWithHoles() throws Exception {
+        testRepeatedLabelsPolygonBorder(squareHoles, "repeatedLabelsAlongLine", "polyHole", 1000);
+    }
+    
+    public void testRepeatedLabelsPolygonBorder(SimpleFeatureSource features, String styleName, String testImageSuffix, int tolerance) throws Exception {
```

Comment:
Trivial: could this be internal / protected / private or named slightly differently to make it clear its a helper not a test?